lib/rubocop/cop/style/to_json.rb

```diff
@@ -0,0 +1,55 @@
+# frozen_string_literal: true
+
+module RuboCop
+  module Cop
+    module Style
+      # Avoids having both `JSON.generate` and `Object#to_json` in your codebase.
+      # They are equivalent, so for simplicity it makes sense to use only one.
+      # `to_json` is also what ActiveModel::Serializer uses, to it's easier to standardize on it.
+      #
+      # @example
+      #   # bad
+      #   JSON.generate(something)
+      #
+      #   # good
+      #   something.to_json
+      class ToJson < Base
+        extend AutoCorrector
+
+        MSG = 'Use `.to_json` instead of `JSON.generate`.'
+
+        # @!method json_generate?(node)
+        def_node_matcher :json_generate?, '(send (const nil? :JSON) :generate ...)'
```

Comment:
This would allow us to also detect `::JSON.generate`:

```suggestion
        def_node_matcher :json_generate?, '(send (const {nil? chase} :JSON) :generate ...)'
```

Reply:
Just noticed I have a typo there: it should be `cbase` not ~`chase`~.

Reply:
What's even weirder is that if I add _this exact file_ as an example in `node_destructuring_spec.rb`, and expect the offense, the spec passes.

```diff
diff --git a/spec/rubocop/cop/internal_affairs/node_destructuring_spec.rb b/spec/rubocop/cop/internal_affairs/node_destructuring_spec.rb
index 72236e565..3a6ae8f87 100644
--- a/spec/rubocop/cop/internal_affairs/node_destructuring_spec.rb
+++ b/spec/rubocop/cop/internal_affairs/node_destructuring_spec.rb
@@ -29,4 +29,67 @@ RSpec.describe RuboCop::Cop::InternalAffairs::NodeDestructuring, :config do
       lhs, rhs = array.children
     RUBY
   end
+
+  context 'example from #12314' do
+    it 'registers an offense' do
+      expect_offense(<<~'RUBY')
+        # frozen_string_literal: true
+
+        module RuboCop
+          ...
+                  add_offense(node) do |corrector|
+                    _receiver, _method_name, *args = *node
+                    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ Use the methods provided with the node extensions instead of manually destructuring nodes.
+                    data, opts = handle_args(args)
+          ...
+        end
+      RUBY
+    end
+  end
 end
```

cc. @koic Any idea why CI didn't catch this?